ci/check-protobuf-against-stable.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+set -euo pipefail
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+readonly SRC_DIR="$(dirname "${SCRIPT_DIR}")"
+
+cd "${SRC_DIR}"
+readonly CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+
+set_latest_stable() {
+  local gitTagSearchSpace
+  if [[ "${CURRENT_BRANCH}" == "main" ]]; then
+    gitTagSearchSpace="--no-merged"
+  else
+    gitTagSearchSpace="--merged"
+  fi
+  LATEST_STABLE="$(git tag ${gitTagSearchSpace} | grep -v "snapshot" | sort -V | tail -1)"
```

Comment:
Bit of a personal preference here, but I'm not too fond of mutating global variables from a function body. You can "return" a string from a function by using `echo` here and then binding it with
```
LATEST_STABLE="$(set_latest_stable)"
```
from the main body.

Reply:
Done, 16158c3